packages/transaction-controller/src/types.ts

```diff
@@ -193,3 +268,243 @@ export type SubmitHistoryEntry = {
+   * have not treated as a special case, such as approve, transfer, and
+   * transferfrom.
+   */
+  contractInteraction = 'contractInteraction',
+
+  /**
+   * A transaction that deployed a smart contract.
+   */
+  deployContract = 'contractDeployment',
+
+  /**
+   * A transaction for Ethereum decryption.
+   */
+  ethDecrypt = 'eth_decrypt',
+
+  /**
+   * A transaction for getting an encryption public key.
+   */
+  ethGetEncryptionPublicKey = 'eth_getEncryptionPublicKey',
+
+  /**
+   * An incoming (deposit) transaction.
+   */
+  incoming = 'incoming',
+
+  /**
+   * A transaction for personal sign.
+   */
+  personalSign = 'personal_sign',
+
+  /**
+   * When a transaction is failed it can be retried by
+   * resubmitting the same transaction with a higher gas fee. This type is also used
+   * to speed up pending transactions. This is accomplished by creating a new tx with
+   * the same nonce and higher gas fees.
+   */
+  retry = 'retry',
+
+  /**
+   * A transaction sending a network's native asset to a recipient.
+   */
+  simpleSend = 'simpleSend',
+
+  /**
+   * A transaction that is signing a message.
+   */
+  sign = 'eth_sign',
+
+  /**
+   * A transaction that is signing typed data.
+   */
+  signTypedData = 'eth_signTypedData',
+
+  /**
+   * A transaction sending a network's native asset to a recipient.
+   */
+  smart = 'smart',
+
+  /**
+   * A transaction swapping one token for another through MetaMask Swaps.
+   */
+  swap = 'swap',
+
+  /**
+   * Similar to the approve type, a swap approval is a special case of ERC20
+   * approve method that requests an allowance of the token to spend on behalf
+   * of the user for the MetaMask Swaps contract. The first swap for any token
+   * will have an accompanying swapApproval transaction.
+   */
+  swapApproval = 'swapApproval',
+
+  /**
+   * A token transaction requesting an allowance of the token to spend on
+   * behalf of the user.
+   */
+  tokenMethodApprove = 'approve',
+
+  /**
+   * A token transaction transferring tokens from an account that the sender
+   * has an allowance of. The method is prefixed with safe because when calling
+   * this method the contract checks to ensure that the receiver is an address
+   * capable of handling the token being sent.
+   */
+  tokenMethodSafeTransferFrom = 'safetransferfrom',
+
+  /**
+   * A token transaction where the user is sending tokens that they own to
+   * another address.
+   */
+  tokenMethodTransfer = 'transfer',
+
+  /**
+   * A token transaction transferring tokens from an account that the sender
+   * has an allowance of. For more information on allowances, see the approve
+   * type.
+   */
+  tokenMethodTransferFrom = 'transferfrom',
+
+  /**
+   * A token transaction requesting an allowance of all of a user's tokens to
+   * spend on behalf of the user.
+   */
+  tokenMethodSetApprovalForAll = 'setapprovalforall',
+}
+
+export type Events = {
```

Comment:
If we're only using this to type the argument in the `swaps` util, then should we remove for now to minimise the code and to keep the the main `hub` consistent?